app/src/main/java/com/mk/tasky/authentication/login/data/LoginRepositoryImpl.kt

```diff
@@ -0,0 +1,21 @@
+package com.mk.tasky.authentication.login.data
+
+import com.mk.tasky.authentication.login.data.mapper.toDomain
+import com.mk.tasky.authentication.login.data.remote.LoginApi
+import com.mk.tasky.authentication.login.data.remote.dto.LoginBodyDto
+import com.mk.tasky.authentication.login.domain.LoginRepository
+import com.mk.tasky.authentication.login.domain.models.LoggedUser
+
+class LoginRepositoryImpl(
+    private val api: LoginApi
+) : LoginRepository {
+    override suspend fun login(email: String, password: String): Result<LoggedUser> {
+        val body = LoginBodyDto(email = email, password = password)
+        return try {
+            val response = api.login(body)
+            Result.success(response.toDomain())
+        } catch (e: Exception) {
+            return Result.failure(e)
```

Comment:
Be careful with catching general exceptions in suspend functions. If the coroutine in which this `login()` function is running is cancelled, the currently suspending function (like `api.login()` here) would throw a `CancellationException` which your catch block would eat up. 
The problem is that due to eating it up, it won't properly be propagated up to the coroutine scope which will therefore not know about the cancellation and potentially resume with code even though it was cancelled. 

Here in this video I talk about it in a bit more detail (starting at 17:33): https://www.youtube.com/watch?v=VWlwkqmTLHc&t=1058s&ab_channel=PhilippLackner

The solution here would be either catching specific exceptions (like `HttpException` and `IOException`) or you check if the exception is a cancellation exception and rethrow it in that case.

Reply:
But in this case, wouldn't it be the same thing?

Ex: If it goes to an exception either HTTP or Cancellation, it would just display an error saying "Unexpected error" or something like that?

Reply:
Already answered in WhatsApp :)